docs/streams/developer-guide/dsl-api.html

```diff
@@ -3134,6 +3192,49 @@ <h4><a id="streams_concepts_globalktable" href="#streams_concepts_globalktable">
                 </div>
             </div>
         </div>
+	
+	<div class="section" id="controlling-update-rate">
+            <span id="streams-developer-guide-dsl-suppression"></span><h2><a class="toc-backref" href="#id32">Controlling KTable update rate</a><a class="headerlink" href="#controlling-update-rate" title="Permalink to this headline"></a></h2>
+	    <p>A KTable is logically a continuously updated table.
+	       These updates make their way to downstream operators whenever new data is available, ensuring that the whole computation is as fresh as possible.
+	       Logically speaking, most programs describe a series of transformations, and the update rate is not a factor in the program behavior.
+	       In these cases, the rate of update is more of a performance concern.
+	       Operators are able to optimize both the network traffic (to the Kafka brokers) and the disk traffic (to the local state stores) by adjusting
+	       the commit interval and batch size configurations.
+	    </p>
+	    <p>However, some applications need to take other actions, such as calling out to external systems, 
+	       and therefore need to exercise some control over the rate of invocations, for example of <pre>KStream#foreach</pre>.
+	    </p>
+	    <p>Rather than achieving this as a side-effect of the  <a class="reference internal" href="memory-mgmt.html#streams-developer-guide-memory-management-record-cache"><span class="std std-ref">KTable record cache</span></a>,
+	       you can directly impose a rate limit via the <pre>KTable#suppress</pre> operator.
+	    </p>
+	    <p>For example:
+	    </p>
+	    <div class="highlight-java"><div class="highlight">
+<pre>
+KGroupedTable&lt;String, String$gt; groupedTable = ...;
+groupedTable
+    .count()
+    .suppress(untilTimeLimit(ofMinutes(5), maxBytes(100_000_000L).emitEarlyWhenFull()))
+    .toStream()
+    .foreach((key, count) -$gt; updateCountsDatabase(key, count));
+</pre>
+	    </div></div>
+	    <p>This configuration ensures that <pre>updateCountsDatabase</pre> gets events for each <pre>key</pre> no more than once every 5 minutes.
+	       Note that the latest state for each key has to be buffered in memory for that 5-minute period.
+	       You have the option to control the maximum amount of memory to use for this buffer (in this case, 100MB).
+	       There is also an option to impose a limit in terms of number of records (or to leave both limits unspecified).
+	    </p>
+	    <p>Additionally, it is possible to choose what happens if the buffer fills up.
+	       This example takes a relaxed approach and just emits the oldest records before their 5-minute time limit to bring the buffer back down to size.
+	       Alternatively, you can choose to stop processing and shut the application down.
+	       This may seem extreme, but it gives you a guarantee that the 5-minute time limit will be absolutely enforced.
```

Comment:
Should we point out, that the feature is incomplete and that 2.2 will have spilling to disk option (plus link to KIP?)

Reply:
We can link to the KIP, I wasn't sure if that was typical.

I'd rather not mention things that might or might not happen in the future. It doesn't seem relevant to the person reading the docs right now.

Reply:
Ack. Yes, we usually link to the KIP page.

Reply:
done.